src/builder/modules/Generator.res

```diff
@@ -143,6 +147,72 @@ let drawLine = (
     Generator_Builder.drawLine(model.contents, from, to, ~color, ~width, ~pattern=[], ~offset=0)
 }
 
+let drawLinePath = (
```

Comment:
Since we already have `drawLine()` and `drawLinePath()` is just an alternative to that, I'm not seeing a compelling need for this function? Keeping the generator as simple as possible is a fundamental principle to help us decide what gets included, and less functions is a key factor for simplicity. Maybe we use it in the Generator_Builder for some purpose, but I'm not sure it needs to be a feature exposed to generator writers?

Reply:
drawLinePath() isn't so much an alternative to drawLine() as it is a shorthand for strings of connected lines- think for outlines of complex shapes. It is also used in the drawRect() function, which in turn is used in drawCuboid().

Reply:
Cool, thanks for explaining.

I do agree that this is a slightly nicer way to handle drawing a line with many points. So the papercraft use case here is to draw a line for a shape with many points (such as a complex shape as you mentioned).

Just a side question - I can't recall a use case for drawing solid line outlines? But you might have something in mind?

Back to the main idea - since we would need an array of points anyway for the shape, we could also just loop over those points and call `drawLine()` right? A loop is a little less convenient, but still trivial to write. And if this use case is rare, then a loop with `drawLine()` should be sufficent?
 
> It is also used in the drawRect() function, which in turn is used in drawCuboid().

`drawRect()` could easily be switched across to use `drawLine()` right? And `drawCuboid()` would not need to change? Is that right? That "offset" chat in the drawLine function might also be relevant here.

So here's my understanding, but correct me if this is not quite right:

* Primary use case is shapes with many points (a few points, such as 4, are easy enough t.o do with drawLine(), so perhaps this helps with shapes with > 4 points), however this use case is not very common.
* This is a little more convenient than `drawLine()` since looping is not required, but `drawLine()` in a loop would achieve the same result, and is trivial to do.

And my other comment:

* I can't recall any use cases for drawing solid outlines, but let me know if you have some in mind.